### PR TITLE
[ci] automatic cherry-picking of branches

### PR DESCRIPTION
**Pull request by nbdd0121**
This adds a label-based automatic cherry-picking (backporting) of pull requests. An example could be seen here: https://github.com/nbdd0121/opentitan/pull/7

This requires GitHub actions to be able to create pull requests, so marking as draft until I get confirmation that this option is enabled.

### CODE REVIEW

File: .github/workflows/cherrypick.yml
Comment (jwnrt) [inline .github/workflows/cherrypick.yml:25]: I'm not sure I understand this condition - won't it be `true` when any pull request is closed regardless of whether it has a `CherryPick:` label?

File: .github/workflows/cherrypick.yml
Comment (jwnrt) [inline .github/workflows/cherrypick.yml:33]: Can't we use `@2.4.1` here?

File: .github/workflows/cherrypick.yml
Comment (jwnrt) [inline .github/workflows/cherrypick.yml:38]: Nit:
```suggestion
            This is an automatic cherry-pick of #${pull_number} to branch `${target_branch}`.
```

File: .github/workflows/cherrypick.yml
Comment (nbdd0121) [reply .github/workflows/cherrypick.yml:25]: Yes. The backport action will check the label to see whether backport is needed and what branch to backport to.

The `label` event is only present when the event is triggered by a label event, so it's not available when the triggering event is a merge.

File: .github/workflows/cherrypick.yml
Comment (nbdd0121) [reply .github/workflows/cherrypick.yml:33]: We can. This is pinning the supply chain security purpose, as this workflow have write access.

**Review by jwnrt — APPROVED** [review]
LGTM as long as we trust this action which has write access to our branches (pinning was a good idea).

**Comment by pamaury** [comment]
No, it looks like a unittest failure due to being a random test, I will look into it, I must have forgotten a corner case. If you re-run it, most likely it will pass.

**Comment by jwnrt** [comment]
I did a surface-level audit of the action and it looks safe. If we want to be extra cautious, I believe there's a GitHub branch protection rule that only allows pull requests to change the `master` branch (if that's not already enabled).

Would be good to clear this PR with a GitHub admin (I think maybe @mundaym and @GregAC).

**Review by mundaym — APPROVED** [review]

**Pull request merged**
